tests/test_util_search.py

```diff
@@ -0,0 +1,175 @@
+            pytest.param("Philadelphia", [], id="no_state_name"),
+        ]
+    )
+    def test_us_state_names(self, search_string, us_state_names):
+        """
+        GIVEN: An input string
+        WHEN:  A LibrarySearchQuery object is instantiated from that string
+        THEN:  The .us_state_names attribute should be list of lowercase state names, for
+               any US states whose name appeared in the search string
+        """
+        sq_obj = LibrarySearchQuery(search_string)
+        assert sq_obj.us_state_names == us_state_names
+
+    @pytest.mark.parametrize(
+        "search_string,library_keywords",
+        [
+            pytest.param("Alpha, Bravo", []),
+            pytest.param("New York Public Library", ["public", "library"]),
+            pytest.param("University of Nebraska", ["university"]),
+        ]
+    )
+    def test_library_keywords(self, search_string, library_keywords):
+        """
+        GIVEN: An input string
+        WHEN:  A LibrarySearchQuery object is instantiated from that search_string
+        THEN:  The .library_keywords attribute should be a list of any tokens which match
+               a word in LibrarySearchQuery.LIBRARY_KEYWORDS
+        """
+        sq_obj = LibrarySearchQuery(search_string)
+        assert sq_obj.library_keywords == library_keywords
+
+    @pytest.mark.parametrize(
+        "search_string,result",
+        [
+            pytest.param("Alpha    \n Bravo  \t Charlie", "Alpha Bravo Charlie", id="extra_whitespace"),
+            pytest.param(
+                ("ABCDE " * (math.floor(LibrarySearchQuery.MAX_SEARCH_STRING_LEN / len("ABCDE ")) + 2)).strip(),
+                ("ABCDE " * math.floor(LibrarySearchQuery.MAX_SEARCH_STRING_LEN / len("ABCDE "))).strip(),
+                id="longer_than_max_len"
+            ),
+            pytest.param(
+                "X" * (LibrarySearchQuery.MAX_SEARCH_STRING_LEN - 2) + ' XXXXX',
+                "X" * (LibrarySearchQuery.MAX_SEARCH_STRING_LEN - 2),
+                id="partial_token"
+            ),
+            pytest.param(1234, '', id="numeric_input"),
+            pytest.param(["a", "b"], '', id="list_input"),
+            pytest.param(None, '', id="none_input"),
+        ]
+    )
+    def test__normalize_search_string(self, search_string, result):
+        """
+        GIVEN: An input string
+        WHEN:  LibrarySearchQuery._normalize_search_string() is called on that string
+        THEN:  A string should be returned which meets these criteria:
+                * Extraneous whitespace characters converted to single spaces
+                * Total length below LibrarySearchQuery.MAX_SEARCH_STRING_LEN
+                * If truncating to fewer total characters than MAX_SEARCH_STRING_LEN,
+                  the string should not end in a partial word.
+        """
+        assert LibrarySearchQuery._normalize_search_string(search_string) == result
```

Comment:
We've got code to correct the most common spelling of "library" ("libary") which isn't tested here; the code may not be in _normalize_search_string but I think it should be.

Reply:
Weird, I remember adding that at some point because I'd seen it in the existing code. I'll make sure it gets back in.